lib/engines/csvEngine.js

```diff
@@ -0,0 +1,163 @@
+var _ = require("lodash");
+var Bromise = require("bluebird");
+var converter = require("json-2-csv");
+var mqtt = require("mqtt");
+var fs = require('fs');
+
+var MQTT_PORT = "1883";
+
+/***
+Config:
+   "generation": {
+        "mode": "csvEngine",
+        "csvEngine": {
+            "fileRepository": ".",
+            "csvOptions": "",
+            "timestamp": 1,
+            "value": 2
+            }
+        }
+***/
+
+function CSVEngine(simulator) {
+
+    this.config = simulator.simulation.generation.csvEngine;
+    this.simulator = simulator;
+    this.mqttServer = "tcp://" + simulator.configuration.server;
```

Comment:
With NodeJS 4.x we can use string templates for this kind of stuff:
```js
this.mqttServer = `tcp://${simulator.configuration.server}`;
```

Reply:
ok